tests/e2e/tests/test_storage.py

```diff
@@ -0,0 +1,146 @@
+        ["k8s", "kubectl", "get", "po", "-n", "kube-system", "-o", "json"],
+        condition=lambda p: "ck-storage" in p.stdout.decode(),
+    )
+    LOG.info("Storage provisioner pod showed up.")
+
+    util.retry_until_condition(
+        h,
+        instance_id,
+        [
+            "k8s",
+            "kubectl",
+            "wait",
+            "--for=condition=ready",
+            "pod",
+            "-n",
+            "kube-system",
+            "-l",
+            "app.kubernetes.io/instance=ck-storage",
+            "--timeout",
+            "180s",
+        ],
+        max_retries=3,
+        delay_between_retries=1,
+        check=True,
+    )
+
+    manifest = MANIFESTS_DIR / "storage-test.yaml"
+    p = h.exec(
+        instance_id,
+        ["k8s", "kubectl", "apply", "-f", "-"],
+        check=True,
+        input=Path(manifest).read_bytes(),
+    )
+
+    LOG.info("Waiting for storage writer pod to show up...")
+    util.retry_until_condition(
+        h,
+        instance_id,
+        ["k8s", "kubectl", "get", "po", "-o", "json"],
+        condition=lambda p: "storage-writer-pod" in p.stdout.decode(),
+        delay_between_retries=10,
+    )
+    LOG.info("Storage writer pod showed up.")
+
+    util.retry_until_condition(
+        h,
+        instance_id,
+        [
+            "k8s",
+            "kubectl",
+            "wait",
+            "--for=condition=ready",
+            "pod",
+            "-l",
+            "k8s-app=storage-writer-pod",
+            "--timeout",
+            "180s",
+        ],
+        max_retries=3,
+        delay_between_retries=1,
+        check=True,
+    )
+
+    LOG.info("Waiting for storage to get provisioned...")
+    util.retry_until_condition(
+        h,
+        instance_id,
+        ["k8s", "kubectl", "get", "pvc", "-o", "json"],
+        condition=lambda p: "myclaim" in p.stdout.decode()
+        and "Bound" in p.stdout.decode(),
+    )
+    LOG.info("Storage got provisioned and pvc is bound.")
+
+    LOG.info("Waiting for storage reader pod to show up...")
+    util.retry_until_condition(
+        h,
+        instance_id,
+        ["k8s", "kubectl", "get", "po", "-o", "json"],
+        condition=lambda p: "storage-reader-pod" in p.stdout.decode(),
+        delay_between_retries=10,
+    )
+    LOG.info("Storage reader pod showed up.")
+
+    util.retry_until_condition(
+        h,
+        instance_id,
+        [
+            "k8s",
+            "kubectl",
+            "wait",
+            "--for=condition=ready",
+            "pod",
+            "-l",
+            "k8s-app=storage-reader-pod",
+            "--timeout",
+            "180s",
+        ],
+        max_retries=3,
+        delay_between_retries=1,
+        check=True,
+    )
+
+    p = h.exec(
+        instance_id,
+        ["k8s", "kubectl", "logs", "storage-reader-pod"],
+        capture_output=True,
+    )
+    assert "test_date" in p.stdout.decode()
+    assert str(datetime.now().year) in p.stdout.decode()
```

Comment:
hmm, could we have a fixed text like "LOREM IPSUM" instead of the current date? should not be a problem, but would keep the logic here simpler, as I don't have to think too much looking at this and the pod template